### PR TITLE
refactor super_node_mediator

### PR DESCRIPTION
**Pull request by dillonroller**

### CODE REVIEW

File: include/am_super/super_node_mediator.h [hunk 1]
```diff
@@ -20,13 +20,9 @@ namespace am
...
 class SuperNodeMediator
 {
-private:
-  // SuperStateMediator super_state_mediator;
-  // AMLifeCycleMediator life_cycle_mediator;
-
 public:
-  SuperNodeMediator();
-  static const std::string SUPER_NODE_NAME; 
+  SuperNodeMediator(std::string node_name = "am_super");
```
Comment (dillonroller) [inline include/am_super/super_node_mediator.h:24]: Node name should always be passed into mediator

File: include/am_super/super_node_mediator.h [hunk 2]
```diff
@@ -196,6 +227,14 @@ class SuperNodeMediator
   /** @return a csv string of names of the manifested nodes not online */
   string manifestedNodesNotOnlineNamesList(Supervisor supervisor);
 
+private:
+  // SuperStateMediator super_state_mediator;
+  // AMLifeCycleMediator life_cycle_mediator;
+
+  bool lifeCycleNotYetImplemented(string node_name);
```
Comment (dillonroller) [inline include/am_super/super_node_mediator.h:234]: move documentation into header file for this function. Mark as deprecated

File: src/am_super/am_super.cpp [hunk 3]
```diff
@@ -538,7 +538,7 @@ class AMSuper : AMLifeCycle
   {
     if(getState() == LifeCycleState::INACTIVE) //if super lifecycle is currently inactive
     {
-      sendLifeCycleCommand(SuperNodeMediator::SUPER_NODE_NAME, LifeCycleCommand::ACTIVATE);
+      sendLifeCycleCommand(node_mediator_.nodeNameStripped(ros::this_node::getName()), LifeCycleCommand::ACTIVATE);
```
Comment (dillonroller) [inline src/am_super/am_super.cpp:541]: Make constant private, reference the name using a mediator function: getName()

Comment (dillonroller) [reply src/am_super/am_super.cpp:541]: Make nodeNameStripper static (or use one that exists already)

**Comment by aroller** [comment]
#70 took care of this.

**Pull request closed without merging**